DV Blog Post

rumackaaron opened the pull request:
So far, just a first draft

rumackaaron committed:
First draft of DV signal

ryantibs commented:
Hi @rumackaaron, a great start!  I'd like let @RoniRos take a pass first, this week, and iterate with you if need.  And then @zackchase can give feedback too.  I'll be able to look back at this next weekend or early next week, then hopefully at that point it'll be pretty much ready-to-go.  That said, several comments from me, to make up front: 

- Needs an abstract (see the way that we did it in the other posts).  The abstract can be very similar to the first paragraph or so.  
- If it makes sense and if relevant, try to mention (and explicitly link to) other blog posts as you're introducing this one.  Probably generally helps with the blog readership (and gives a sense of connectedness).
- I'd be more explicit with the setup: when did this signal "begin"?  How many counties do we get coverage on (i.e., do we have estimates for) per day?  How many claims are typically present in a given day (can we even share the rough order of magnitude? we say 10-15% of the US, so in principle, if somebody knew the total number of claims across all health systems in a given day on average, they could calculate this, so I'm not sure why it would be bad to state it).
- I might suggest an example earlier on, to draw the reader's attention in.
- I'd be more explicit with some details on missingness: what's the min-samples-needed for an estimate?
- Generally, I'm not sure why we need to give mathematical details on the weekday correction---I think these may be a bit distraction.
- When we say that backfill causes us to underestimate the signal, I think we should soften the language here: it can definitely go the other way.  See https://cmu-delphi.github.io/covidcast/talks/rmls20/talk.html#(25) (example due to Jacob Bien originally).  Feel free to grab that example by the way, for the post, if you want it (code is in the talk).
- We introduce the hospital admissions indicator early on in the post, but then I don't see anything about it?
- You should contribute `fips_to_hhs()` to the covidcast package!  (The earlier you do it, the earlier we can merge the PR, and you can just call this function from the package for your post.)
- The sensorization is cool ... I'm tempted to suggest we write a separate post spelling that out much more explicitly (and probably doing more analysis).  But I guess a teaser wouldn't hurt here?  I would be sure to refer to our sensor fusion systems, cite David's thesis  (Farrow 2016) and our sensor fusion paper (Jahja et al. 2019), which rely on this idea.  It would strengthen the argument.

rumackaaron committed:
Incorporated Ryan's suggestions

ryantibs commented:
Thanks @rumackaaron.  Sorry to be annoying but @RoniRos any progress here on your pass?  You can either take a pass directly and make a commit, or leave comments through GitHub's review feature, whatever's easiest for you.

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+---
+title: "Using Electronic Health Records to Develop Real-Time COVID-19 Indicators"
```
```suggestion
title: "A Syndromic COVID-19 Indicator Based on Insurance Claims of Outpatient Visits"
```

If "syndromic" is not familiar enough to the intended readers, we can try "Symptom-based", or drop this modifier altogether.

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+    code_folding: hide
+---
+
+In addition to symptom surveys, one of our most prized data sources comes from health insurance claims data. Healthcare partners provide us with the counts of people going to the doctor due to COVID-like illness in counties across the entire country. We use this data to calculate a signal that estimates the level of COVID-like illness in every county where enough data is available. In this post,  ...
```
```suggestion
In addition to symptom surveys, one of our most prized data sources is health insurance claims data. Healthcare partners provide us with daily counts of people who visited a doctor due to COVID-like illness in counties across the entire country. We use this data to calculate a signal that estimates the level of COVID-like illness in every county where enough data is available. In this post, we discuss the data itself, what this data can tell us that the number of confirmed cases can't, and how we calculate a %CLI signal. We also investigate why the data appeared to become much less useful over time, and how we discovered that the data just needed a simple adjustment!
```

RoniRos commented:
> Thanks @rumackaaron. Sorry to be annoying but @RoniRos any progress here on your pass? You can either take a pass directly and make a commit, or leave comments through GitHub's review feature, whatever's easiest for you.

I finally got to it today, should be able to finish either tonight or tomorrow mid-day.
I am using GitHub's review feature. 
@ryantibs quick question:  how do I 'preview' the the draft blog post as it is now?  the only 'preview' tab I see is for my comment, like this one.

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+<!--more-->
+
+In previous blog posts, we have written about indicators generated from surveys that we run through [Facebook](https://delphi.cmu.edu/blog/2020/08/26/covid-19-symptom-surveys-through-facebook/) and [Google](https://delphi.cmu.edu/blog/2020/09/18/covid-19-symptom-surveys-through-google/). In addition to these, we have also generated indicators from electronic medical records. We received access to ...
```
Do we want to discuss the hospital records?  (1) we were not supposed to receive them based on our DUA; and (2) we no longer receive them.  The alternative is to discuss the insurance claims only.  If we want to show some results that involve the hospital records, we can add a footnote that these results are based on both insurance claims and hospital records (to which we had access until October 1).
If we decide to continue to discuss hospital records, note that they do NOT cover 10%-15% of the US population, but rather only about 40% of that 10%-15%, namely 4%-6%.

RoniRos replied:
```suggestion
In previous blog posts, we have written about indicators generated from surveys that we run through [Facebook](https://delphi.cmu.edu/blog/2020/08/26/covid-19-symptom-surveys-through-facebook/) and [Google](https://delphi.cmu.edu/blog/2020/09/18/covid-19-symptom-surveys-through-google/). In addition to these, we have also generated indicators from electronic medical records. We received access to hospital records and insurance claims data from healthcare partners covering 10-15% of the population of the United States. One indicator that we created from the outpatient insurance claims portion of the data is the "Doctor Visits" signal, which estimates the percentage of office visits due to COVID-like illness.
```

ryantibs commented:
@RoniRos @rumackaaron To preview it, Aaron's would have to build it (knit it) into HTML and commit and push this to the branch.  His commits only show the markdown source though, so it doesn't look like he did this.  

If you want, I can do it later.  The blocker may be that you'll actually have to pull the branch to your local machine and view the HTML on your machine.  Usually, GitHub doesn't allow you to actually look at HTML files through the browser.  So if you've only been using GitHub through the browser so far, then this will be the blocker ... of course someone could just email/slack you the HTML file.

ryantibs commented:
@RoniRos Very important comment (in case you were unaware): this is a public repo.  Anything you say here (comments about the data, etc.) will be open to the world ...

capnrefsmmat committed:
Merge branch 'main' into dv-signal

capnrefsmmat commented:
[Did someone say preview?](http://rosmarus.refsmmat.com/blog/2020/10/14/using-electronic-health-records-to-develop-real-time-covid-19-indicators/)

RoniRos commented:
> [Did someone say preview?](http://rosmarus.refsmmat.com/blog/2020/10/14/using-electronic-health-records-to-develop-real-time-covid-19-indicators/)

Thanks @capnrefsmmat .  That helps.  I'm still not happy with the fact that this drafting/reviewing process is public -- can't we do better?  Do it on a private repo?

capnrefsmmat commented:
A private repo possible, with complications and significant inconvenience.

I think transparency about our data, analyses, and even blog posts is a laudable thing when our work will be under so much public scrutiny. Revision in the open doesn't worry me too much, unless we accidentally share data restricted by DUA or something like that.

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+In previous blog posts, we have written about indicators generated from surveys that we run through [Facebook](https://delphi.cmu.edu/blog/2020/08/26/covid-19-symptom-surveys-through-facebook/) and [Google](https://delphi.cmu.edu/blog/2020/09/18/covid-19-symptom-surveys-through-google/). In addition to these, we have also generated indicators from electronic medical records. We received access to ...
+
+Electronic health records are extremely useful in providing information about epidemics in real time. A patient may interact with the healthcare system days before receiving a test (if any), and several more days before receiving a positive result and being reported as a confirmed case by a state's Department of Health. This can make our signal an **early indicator** of confirmed cases, and certa ...
```
```suggestion
Electronic health records are extremely useful in providing information about epidemics in near real time. A patient may interact with the healthcare system days before receiving a test (if any), and several more days before receiving a positive result and being reported as a confirmed case by a state's Department of Health. This can make our signal an **early indicator** of confirmed cases, and even more so an early indicator of deaths. Early knowledge of an increase in people with COVID-like symptoms is extremely important, as early containment can help contain an outbreak.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+Electronic health records are extremely useful in providing information about epidemics in real time. A patient may interact with the healthcare system days before receiving a test (if any), and several more days before receiving a positive result and being reported as a confirmed case by a state's Department of Health. This can make our signal an **early indicator** of confirmed cases, and certa ...
+
+Additionally, the reported case counts have other issues:
```
```suggestion
The value of insurance claims data is even more apparent when you consider other problems with official case counts:
```

zackchase commented:
What precisely is the inconvenience? I don't think its a huge deal but all
else equal I’m with Roni on this one, don’t see much benefit to our anal
detailed comments during drafting being aired in public.

On Sat, Oct 24, 2020 at 8:03 PM RoniRos <notifications@github.com> wrote:

> *@RoniRos* commented on this pull request.
> ------------------------------
>
> In content/post/2020-10-14-dv-signal.Rmd
> <https://github.com/cmu-delphi/delphi-blog/pull/34#discussion_r511527657>:
>
> > +date: 2020-10-14
>
> +tags: ["nowcasting", "data sources"]
>
> +output:
>
> +  html_document:
>
> +    code_folding: hide
>
> +---
>
> +
>
> +In addition to symptom surveys, one of our most prized data sources comes from health insurance claims data. Healthcare partners provide us with the counts of people going to the doctor due to COVID-like illness in counties across the entire country. We use this data to calculate a signal that estimates the level of COVID-like illness in every county where enough data is available. In this post, we discuss the data itself, what this data can tell us that the number of confirmed cases can't, and how we calculate a %CLI signal. We also investigate why the data appeared to become much less useful over time, and how we discovered that the data just needed a simple adjustment!
>
> +
>
> +<!--more-->
>
> +
>
> +In previous blog posts, we have written about indicators generated from surveys that we run through [Facebook](https://delphi.cmu.edu/blog/2020/08/26/covid-19-symptom-surveys-through-facebook/) and [Google](https://delphi.cmu.edu/blog/2020/09/18/covid-19-symptom-surveys-through-google/). In addition to these, we have also generated indicators from electronic medical records. We received access to hospital records and insurance claims data from healthcare partners covering 10-15% of the population of the United States. One indicator that we created from this data is the "Doctor Visits" signal, which estimates the percentage of office visits due to COVID-like illness.
>
> +
>
> +Electronic health records are extremely useful in providing information about epidemics in real time. A patient may interact with the healthcare system days before receiving a test (if any), and several more days before receiving a positive result and being reported as a confirmed case by a state's Department of Health. This can make our signal an **early indicator** of confirmed cases, and certainly an early indicator of deaths. Early knowledge of an increase in people with COVID-like symptoms is extremely important, as early containment can prevent an outbreak.
>
> +
>
> +Additionally, the reported case counts have other issues:
>
>
> ⬇️ Suggested change
>
> -Additionally, the reported case counts have other issues:
>
> +The value of insurance claims data is even more apparent when you consider other problems with official case counts:
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cmu-delphi/delphi-blog/pull/34#pullrequestreview-516314817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AASHRTRV2RDID4GPNJYYH6LSMNTO3ANCNFSM4SSTKQGQ>
> .
>

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+Additionally, the reported case counts have other issues:
+
+* Limited testing capacity: Especially prevalent in the earlier days of the pandemic, limited testing capacity means that there were many people who saw a doctor for COVID-like symptoms but never even received a test. These people would be present in our data but absent from the confirmed counts.
+* Backlogs: A patient can be first reported as a confirmed case can be several days after they were tested. Since our data is linked to the date that the patient interacted with the healthcare system, we report a more accurate date.
+* Revisions: Some states have changed their definition of which tests indicate a confirmed COVID-19 case. This can be done retroactively, including hundreds or thousands of past cases and reporting them as "new" on the day on which they changed their definition. These cases are clearly assigned the wrong date, but we have no way of determining when the test was actually taken. Our data is subject ...
```
```suggestion
* Limited testing capacity: Especially prevalent in the earlier days of the pandemic, limited testing capacity means that there were many people who saw a doctor for COVID-like symptoms but never received a test even though they should have. These people would be present in our data but absent from the confirmed counts.
* Reporting delays:  From the moment a patient is first seen or evaluated, it may take several days (especially in the early days of the pandemic) until their specimen is collected.  It then takes a few more days for the test to be run.  Next, if the test is positive it is reported to the health authorities as required by law.  The health authorities typically do further verification and investigation (for example, determining age and other demographics) before adding the case into their public website.  As a result, many patients are reported as confirmed cases several weeks after they were first medically attended.  While insurance claims also have built-in delays (see below), a meaningful claims-based signal can be estimated within 3-4 days of the doctor's visit.
* Dating accuracy: Cumulative confirmed cases scraped and reported by sites such as JHU and USA Facts (and mirrored by COVIDcast) are based on the date the case counts were publicly posted.  As a result, the "number of new daily counts" derived by successive subtraction from these reports refers to the number of cases _published_ on that date, not those that were _lab-confirmed_ on that date.  In contrast, insurance claims records indicate the "date of service" -- the date that the patient interacted with the healthcare system, which is typically closer to the testing date.  As explained above, there can be a gap of up to several weeks between these two dates.
* Retrospective revisions: Some states have changed their definition of confirmed COVID-19 case, sometimes. This is sometime done retroactively, adding hundreds or thousands of past cases and reporting them as "new" on the day on which the definition was changed. These cases are clearly assigned the wrong date, but we have no way of determining when the test was actually taken.  Insurance claims data is not subject to such revisions.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+* Backlogs: A patient can be first reported as a confirmed case can be several days after they were tested. Since our data is linked to the date that the patient interacted with the healthcare system, we report a more accurate date.
+* Revisions: Some states have changed their definition of which tests indicate a confirmed COVID-19 case. This can be done retroactively, including hundreds or thousands of past cases and reporting them as "new" on the day on which they changed their definition. These cases are clearly assigned the wrong date, but we have no way of determining when the test was actually taken. Our data is subject ...
+
+### Doctor Visits Indicator
```
```suggestion
### The _Doctor Visits_ (DV) Indicator
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+### Doctor Visits Indicator
+The Doctor Visits indicator is based solely on insurance claims. For a given geographic area and a given day, we receive the count of claims that fall into each of five categories:
+
+1. Total: All claims
```
```suggestion
1. Total: All claims, whether or not related to COVID-19.  This is also known as 'all-cause' claims.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+The Doctor Visits indicator is based solely on insurance claims. For a given geographic area and a given day, we receive the count of claims that fall into each of five categories:
+
+1. Total: All claims
+2. Flu: ICD-10 primary code starting with J09, J10, or J11. This is an explicit diagnosis of flu.
```
```suggestion
2. "Flu": ICD-10 primary code starting with J09, J10, or J11. This is an explicit diagnosis of flu.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+1. Total: All claims
+2. Flu: ICD-10 primary code starting with J09, J10, or J11. This is an explicit diagnosis of flu.
+3. COVID-like: ICD-10 primary code in {U07.1, U07.2, B97.29, J12.81, Z03.818, B34.2, J12.89}. Some of these codes correspond to an explicit diagnosis of COVID-19, while others indicate similar symptoms.
```
```suggestion
3. "COVID-like": ICD-10 primary code in {U07.1, U07.2, B97.29, J12.81, Z03.818, B34.2, J12.89}. Some of these codes correspond to an explicit diagnosis of COVID-19, while others indicate similar symptoms.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+1. Total: All claims
+2. Flu: ICD-10 primary code starting with J09, J10, or J11. This is an explicit diagnosis of flu.
+3. COVID-like: ICD-10 primary code in {U07.1, U07.2, B97.29, J12.81, Z03.818, B34.2, J12.89}. Some of these codes correspond to an explicit diagnosis of COVID-19, while others indicate similar symptoms.
+4. Flu-like: ICD-10 primary code of J22 (acute lower respiratory infection) or B34.9 (viral infection, unspecified).
```
```suggestion
4. "Flu-like": ICD-10 primary code of J22 (acute lower respiratory infection) or B34.9 (viral infection, unspecified).
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+2. Flu: ICD-10 primary code starting with J09, J10, or J11. This is an explicit diagnosis of flu.
+3. COVID-like: ICD-10 primary code in {U07.1, U07.2, B97.29, J12.81, Z03.818, B34.2, J12.89}. Some of these codes correspond to an explicit diagnosis of COVID-19, while others indicate similar symptoms.
+4. Flu-like: ICD-10 primary code of J22 (acute lower respiratory infection) or B34.9 (viral infection, unspecified).
+5. Mixed: ICD-10 primary code of Z20.828 (suspected exposure to COVID-19) or J12.9 (viral pneumonia).
```
```suggestion
5. "Mixed": ICD-10 primary code of Z20.828 (suspected exposure to COVID-19) or J12.9 (viral pneumonia).
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+5. Mixed: ICD-10 primary code of Z20.828 (suspected exposure to COVID-19) or J12.9 (viral pneumonia).
+
+
+We calculate the percent of COVID-like illness (%CLI) as (COVID-like + Flu-like + Mixed - Flu) / Total. We subtract the Flu counts because they are unrelated to coronavirus. The indicator is available at four geographic levels: county, MSA, HRR, and state. See [here](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/doctor-visits.html) for more details.
```
```suggestion
We estimate the percent of COVID-like illness (%CLI) as (COVID-like + Flu-like + Mixed - Flu) / Total. We subtract the Flu counts because, although they are symptomatically COVID-like, they are known to be unrelated to coronavirus, thus making the signal more directly related to COVID-19. The indicator is available at four geographic levels: county, MSA, HRR, and state. See [here](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/doctor-visits.html) for more details.
```

@mariajahja Did we ever resolved why we are subtracting 'flu'?  Given that on any date, any patient is recorded in at most one of these four categories, it would seem that we need to exclude, but not subtract, the confirmed-flu cases.  And yet maybe subtraction led to better correlations?

mariajahja replied:
We originally thought to construct the signal on the "rate" scale rather than "counts". Therefore, we were subtracting the true flu rate from the mixed+flu_like rate, where the latter was thought to have counted patients that had flu but weren't definitive. But yes, the way these counts were constructed should ensure that this doesn't happen. The implementation does work on the (smoothed and padded) rate scale, rather than counts. 

Note that during the summer, the 'flu' category is essentially nil, and so this subtraction didn't do anything. Earlier on in March with a non-zero flu signal, the subtraction looked necessary to avoid having a strong corona signal that we thought was impossible. I assume that coding practices have changed since then, with doctors able to discern more clearly between covid and flu, and so for the upcoming flu season it's not clear to me whether this subtraction is helpful or necessary.

RoniRos replied:
Thanks @mariajahja .
> the subtraction looked necessary to avoid having a strong corona signal that we thought was impossible.

This suggests to me a (admittedly post hoc) explanation:  the more confirmed-flu there is, the higher the expected fraction of flu-like that is due to flu.  If true, then rather than subtracting, we should train a sensor on these 4 covariates, and make it an adaptive sensor so it continuously adapts to the most recent coding practices.   @rumackaaron @ryantibs

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+We calculate the percent of COVID-like illness (%CLI) as (COVID-like + Flu-like + Mixed - Flu) / Total. We subtract the Flu counts because they are unrelated to coronavirus. The indicator is available at four geographic levels: county, MSA, HRR, and state. See [here](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/doctor-visits.html) for more details.
+
+The signal is available daily starting February 1, 2020 (although understandably, the signal is nearly zero in all locations until mid-March). To preserve privacy and data integrity, we do not report a signal for a given location and date if there are less than 500 total visits in the last seven days before that date. Following these restrictions, we are able to produce estimates for about 2000 c ...
```
```suggestion
The signal is available daily starting February 1, 2020 (although understandably, it is nearly zero in all locations until mid-March). To preserve privacy and data integrity, we do not report a signal for a given location and date if there are fewer than 500 total visits in the seven days ending on that date. Following these restrictions, we are able to produce each day estimates for about 2000 counties, about two-thirds of the counties in the U.S. and accounting for over 90% of the country's population.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
````diff
@@ -0,0 +1,259 @@
+```
+
+### Backfill
+Electronic medical records have a few interesting challenges not present in the survey data. First, is that the data is available at a significant and variable latency (we call this "backfill"). For claims related to a given day, we will receive only about half of them within the first seven days after that date. Due to backfill, we cannot make estimates of our indicator for a given day until fou ...
````
```suggestion
Medical insurance claim records have a few interesting challenges not present in the survey data. First, the data is available at a significant and variable latency, often referred to as "backfill".  Due to backfill, we cannot reliably estimate our indicator for a given date of service until four days have passed.  We continue to revise the indicator's value every day thereafter based on newly received claims.  For a given date of service, we typically receive only about half of the total claims within seven days following that date.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+Interestingly, certain types of claims have a longer latency than others. We found that for a given date, as more claims are available, %CLI usually increases. Covid-like claims tend to have a longer latency and thus our early estimates of the indicator usually underestimate the final value.
+
+### Weekday Effects
+Another challenge is the influence of the day of the week on %CLI. On weekends, both total counts and COVID-like counts decrease, but proportionally, total counts decrease more. This is because people are more likely to delay seeking medical care for non-acute issues than acute issues. The total counts include many visits related to non-acute issues, but almost all COVID-like counts are due to ac ...
```
```suggestion
Another challenge is the influence of the day of the week on %CLI. On weekends, both total counts and COVID-like counts decrease, but proportionally, total counts decrease more. This is because doctor visits during the weekend tend to focus on acute care.  The total counts include many visits related to non-acute issues, but almost all COVID-like counts are due to acute issues. Without correcting for this weekday effect, the %CLI curve has a "sawtooth" pattern, spiking on weekends. We derived a method to create a corrected signal that accounts for this weekday effect (for a formal mathematical description of the method, see our [API documentation](https://cmu-delphi.github.io/delphi-epidata/api/covidcast-signals/doctor-visits.html#day-of-week-adjustment)). Below, we see the effect of making these corrections. When we do not correct for the weekday effect, we see a sawtooth pattern that clearly does not represent true changes in COVID-like illness within a location. However, after making the weekday correction, we get a smooth curve that looks reasonable. It is important to note that this correction is not time-wise smoothing! Rather, we are making a point-wise correction based on seven fitted parameters, one for each day of the week.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
````diff
@@ -0,0 +1,259 @@
+```
+
+### Evaluating the Indicator
+We can evaluate the Doctor Visits indicator by measuring its correlation with cases. We hope that when the Doctor Visits indicator is high, cases are also high. We can measure this correlation across two axes: space and time Let $c_{it}$ and $d_{it}$ be the confirmed COVID-19 rate and DV indicator respectively, at location $i$ and time $t$. The space-wise correlation of an individual location $i$ ...
````
Do you want to also show the average time-wise correlation over all counties with cumulative cases > N (or with population > N) as a function of N?  This will show whether the cases of poor correlation are due to small counts (less differentiates power).

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+  return(cor(x, y, method="spearman",use="na.or.complete"))
+}
+
+ggplot(inner_join(dv_cases_df,lm_dv_cases_df, by="geo_value") %>%
```
These plots are not showing in the HTML that @capnrefsmmat produced.

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,259 @@
+
+We would also like to see how this data compares to that of other healthcare insurance providers. Some insurance providers cover different populations with different rates, which could make their signal more or less correlated with the reported case incidence. We would expect that by combining datasets and covering more people, we can produce a more useful signal.
+
+We believe that this dataset will be useful for forecasting Covid cases and deaths, since a doctor visit usually precedes a confirmed test and death. We can compare time-wise and space-wise correlations for cases lagged by a week or two, as well as deaths. Forecasting models can also take the current DV signal as a feature to help predict what the case incidence will be several weeks into the fut ...
```
Comparing lagged correlations is simple enough that we should do it and add it to the post.

RoniRos reviewed (commented):
LGTM.

My original intent was actually for this post to be about the peculiarities of electronic health records, based on the presentation I had given (and put in the Presentations folder).  This post is really about the DV signal.  But I think that's actually good: we can write a separate post at some point soon about our Hospital Admissions signal, and maybe later on we can go back to the idea of a general post about EHRs, e.g. after we incorporate CHC.

rumackaaron committed:
Apply suggestions from Roni

Co-authored-by: RoniRos <roni+github@cs.cmu.edu>

rumackaaron committed:
Lagged correlations and threshold correlations

rumackaaron commented:
HTML files are included in the `.gitignore` file, and cannot be attached to this thread, so I will upload the latest HTML to Slack for convenience. I added the following plots:

- average time-wise correlation over all counties with cumulative cases > N
- lagged correlations
- fixed the plots that were not showing up in the HTML

I'm happy with the blog post, I just think that it should be acknowledged that Maria put a ton of work into every phase of this project, including writing almost all of the code.

ryantibs commented:
@rumackaaron @RoniRos It's looking great.  

Two things that are still needed right away: 1. add Roni as an author (and add this one-line bio, below Aaron's), and 2. add an acknowledgements paragraph you can see how we've done it in other posts.  The acknowledgements paragraph can make it clear that Maria did a ton of the work on the implementation side. 

@zackchase @Akvannortwick I know you guys are taking a pass for style, writing, etc.  The content overall looks very good to me.   I'll still want to take one more pass over it myself for content tweaks.  For example, I want to suggest to we get rid of the "cheating" correction in the "Correlation Deterioration" section (blue line in the second to last figure), and get rid of the last section "Lagging or Leading" entirely.  

For now, for efficiency, I say we parallel process: I will follow-up with Aaron + Roni on slack to discuss, and Zack + Kari you should take your pass right away, but just don't worry about the last section at all (and worry about the part discussing the "cheating" correction).

rumackaaron committed:
Acknowledgements and Roni as author

rumackaaron committed:
Three suggestions from Ryan

rumackaaron committed:
Refactor to use covidcast_cor

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,286 @@
+    code_folding: hide
+---
+
+In addition to symptom surveys, one of our most prized data sources is health insurance claims data. Healthcare partners provide us with daily counts of people who visited a doctor due to COVID-like illness in counties across the entire country. We use this data to calculate a signal that estimates the level of COVID-like illness in every county where enough data is available. In this post, we di ...
```
```suggestion
In addition to symptom surveys, one of our most prized data sources is health insurance claims data. Healthcare partners provide us with daily counts of people who visited a doctor due to COVID-like illness in counties across the entire country. We use this data to calculate a signal that estimates the level of COVID-like illness in every county where enough data is available. In this post, we discuss the data itself, what this data can tell us that the number of confirmed cases can't, and how we calculate a %CLI signal. We also investigate why the data appeared to become much less useful over time, and how we discovered that the data just needed a simple adjustment!  Of note, our volume of health insurance claims data is about to grow dramatically thanks to the generosity of Change Healthcare, who has donated their data to us.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,286 @@
+
+<!--more-->
+
+In previous blog posts, we have written about indicators generated from surveys that we run through [Facebook](https://delphi.cmu.edu/blog/2020/08/26/covid-19-symptom-surveys-through-facebook/) and [Google](https://delphi.cmu.edu/blog/2020/09/18/covid-19-symptom-surveys-through-google/). In addition to these, we have also generated indicators from electronic medical records. We received access to ...
```
```suggestion
In previous blog posts, we have written about indicators generated from surveys that we run through [Facebook](https://delphi.cmu.edu/blog/2020/08/26/covid-19-symptom-surveys-through-facebook/) and [Google](https://delphi.cmu.edu/blog/2020/09/18/covid-19-symptom-surveys-through-google/). In addition to these, we have also generated indicators from electronic medical records. We received access to hospital records and insurance claims data from healthcare partners covering 10-15% of the population of the United States. One indicator that we created from the outpatient insurance claims portion of the data is the "Doctor Visits" signal, which estimates the percentage of office visits due to COVID-like illness. We have also partnered with Change Healthcare to receive insurance claims data, and we are in the process of creating a similar signal from their data. Their dataset covers approximately 45% of the population of the United States, which we expect to significantly boost the accuracy and resolution of a "Doctor Visit" signal.
```

RoniRos commented on content/post/2020-10-14-dv-signal.Rmd:
```diff
@@ -0,0 +1,286 @@
+### Open Questions and Future Directions
+We were happy to see that a location-specific correction was able to greatly increase the space-wise correlation. We have seen that the regression coefficients of the correction, which are fit using a sliding window, change significantly over time. We would like to look into this non-stationary relationship between the DV signal and case incidence to better understand why it changes, and why coun ...
+
+We would also like to see how this data compares to that of other healthcare insurance providers. As mentioned above, we have received access to data from Change Healthcare, which has a larger coverage than the provider of this data. We would be interested in comparing Change's signal to this signal in sample size, correlations, and behavior. Some insurance providers cover different populations w ...
```
```suggestion
We would also like to see how the accuracy and coverage of EMR-based signals can be increased when data coverage increases dramatically. As mentioned above, we have received access to data from Change Healthcare, which when fully extracted will increase our coverage five fold -- to over 50% of the U.S. population. We would be interested in comparing Change's signal to our current signal in sample size, correlations, and behavior. Some insurance providers cover different populations with different rates, which could make their signal more or less correlated with the reported case incidence. We would expect that by combining datasets and dramatically increasing coverage we can produce a significantly more useful signal.
```

RoniRos reviewed (commented):
Adding (& editing) shutouts for CHC.

rumackaaron committed:
Apply Roni's suggestions

Co-authored-by: RoniRos <roni+github@cs.cmu.edu>

Akvannortwick committed:
DV Edits with Zach

capnrefsmmat committed:
Merge branch 'dv-signal' of github.com:cmu-delphi/delphi-blog into dv-signal

capnrefsmmat committed:
Don't include links in the teaser

capnrefsmmat committed:
Fix broken Markdown formatting

capnrefsmmat committed:
Improve code formatting and cache chunks

capnrefsmmat committed:
Formatting improvements

capnrefsmmat committed:
Minor edits

Akvannortwick committed:
Suggested Edits from Alex

ryantibs committed:
Final version of DV post

- Text finalized through GDoc edits with Roni and Zack
- Code tweaks mainly to make the figures larger

capnrefsmmat committed:
Merge branch 'main' into dv-signal

capnrefsmmat committed:
Retag posts

capnrefsmmat committed:
Finalize DV post and render HTML
